File: plugins/OrchidChangelog/src/main/java/com/eden/orchid/languages/model/ChangelogModel.java
package com.eden.orchid.languages.model;

import lombok.Getter;

import javax.inject.Singleton;
import java.util.List;

@Singleton
public class ChangelogModel {

    @Getter
    private String sourceUrl;

    @Getter
    private List<ChangelogVersion> versions;

    public void initialize(List<ChangelogVersion> versions, String sourceUrl) {
        this.sourceUrl = sourceUrl;
        this.versions = versions;
    }

}

File: plugins/OrchidChangelog/src/main/kotlin/com/eden/orchid/changelog/model/ChangelogModel.kt
package com.eden.orchid.changelog.model

import com.eden.orchid.api.generators.OrchidCollection
import com.eden.orchid.api.generators.OrchidGenerator
import com.eden.orchid.api.theme.pages.OrchidPage
import javax.inject.Singleton

class ChangelogModel(
    val versions: List<ChangelogVersion>
) : OrchidGenerator.Model {

    override val allPages: List<OrchidPage> = emptyList()
    override val collections: List<OrchidCollection<*>> = emptyList()

    fun getVersion(versionName: String): ChangelogVersion? {
        return versions.find { it.version == versionName }
    }
}

